### DISCUSSION TITLE
Feature Request: different destinations based upon log_level

### DISCUSSION THREAD

**Discussion by MadBomber** [reidmorrison/semantic_logger#198] [Q&A] [answered]
I'm looking at updating my standard logging capability.  I like the looks of the semantic logger.  I also like the feature in Yell that allows me to specify different log destinations on a per log-level basis.  For example I could define a destination for all `trace` log entries to go to a file named trace.logm  all `info` and 'debug' log entries to go to STDOUT, and all `error` and above log entries to go to a centralized log service.

Does anyone else think that would be a handy addition to SemanticLogger?

**Comment by reidmorrison** [comment]
Use the filter attribute when adding the appender to filter out any log_level you don't want sent to that appender.
Filter Documentation: https://rocketjob.github.io/semantic_logger/filtering.html
All the fields that can filtered on: https://rocketjob.github.io/semantic_logger/log_struct.html

**Comment by MadBomber** [comment]
I will review that capability.  Thank you for pointing it out.  To be clear I'm looking to spin up one logger instance `logger` where logger.info goes to STDOUT and logger.debug goes to STDERR and everything else goes to a file.

**Comment by reidmorrison** [comment]
Filtering supports the above use-case.

**Comment by MadBomber** [comment, marked as answer]
For the next guy/gal ... filtering is easy.  semantic_logger is now my favorite logger libraru!
```
# level_index #=> 0     1     2    3    4     5
level_names = %w[ trace debug info warn error fatal ]

level_names.each_with_index do |level_name, level_index|
  SemanticLogger.add_appender(
    file_name:  "#{level_name}.log",
    filter:     Proc.new { |log| level_index == log.level_index }
  )
end
```
